packages/tooltip/src/Tooltip.spec.tsx

```diff
@@ -53,6 +65,67 @@ describe('packages/tooltip', () => {
       expect(tooltip).not.toBeVisible();
     });
 
+    test('when "triggerEvent" is set to "hover", hovering on and off the trigger opens and closes the tooltip', async () => {
+      const { getByTestId, queryByTestId, button } = renderTooltip({
+        triggerEvent: 'hover',
+      });
+
+      fireEvent.mouseEnter(button);
+
+      await waitFor(() => getByTestId(tooltipTestId), { timeout: 500 });
+
+      expect(getByTestId(tooltipTestId)).toBeInTheDocument();
+
+      fireEvent.mouseLeave(button);
+
+      await waitForElementToBeRemoved(getByTestId(tooltipTestId), {
+        timeout: 500,
+      });
+
+      expect(queryByTestId(tooltipTestId)).not.toBeInTheDocument();
+    });
+
+    const fireEventMap = {
+      hover: {
+        on: fireEvent.mouseEnter,
+        off: fireEvent.mouseLeave,
+      },
+      click: {
+        on: fireEvent.click,
+        off: fireEvent.click,
+      },
+    };
+
+    async function testTriggerEventWhenDisabled(
+      triggerEvent: 'hover' | 'click',
+    ) {
+      test(`when triggerEvent is "${triggerEvent}"`, async () => {
+        const { queryByTestId, button } = renderTooltip({
+          triggerEvent,
+          enabled: false,
+        });
+
+        fireEventMap[triggerEvent].on(button);
+
+        // Wait for 200ms to allow the component to animate out and remove the element from the DOM
+        await waitForTimeout(200);
+
+        expect(queryByTestId(tooltipTestId)).toBeFalsy();
```

Comment:
?

```suggestion
        expect(queryByTestId(tooltipTestId)).not.toBeInTheDocument();
```

Ditto below.

Reply:
In this case they are functionally identical – queryByTestId returns null when the element is not in the document. Still, I'll update for consistency's sake.

Reply:
Yeah I know. The benefit of this named expectation is two fold:

- Is clearer to the reader what you're asserting, so easier to maintain as you know what you're testing for
- If the expectation is not met then the error message given by react-testing-library is clearer.